File: 剑指Offer/15.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        if n == 0:
            return 1;
        
        if n < 0:
            return 1 / myPow(x, -n)
        
        temp = self.myPow(x, int(n / 2))
        if n & 1 :
            return temp * temp * x
        else:
            return temp * temp

File: 剑指Offer/15.cpp
//采用分治法处理；

class Solution {
public:
    double myPow(double x, int n) {
        if (n == 0){
            return 1;
        }

        double temp = myPow(x, n / 2);
        double result = temp * temp;
        if (n & 1){ //采用位运算的方式判断是奇数还是偶数；
            if (n < 0) result /= x;
            else result *= x;
        }

        return result;
    }
};